nntrainer/compiler/ini_interpreter.cpp

```diff
@@ -0,0 +1,335 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   ini_interpreter.cpp
+ * @date   02 April 2021
+ * @brief  NNTrainer Ini Interpreter (partly moved from model_loader.c)
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jijoong Moon <jijoong.moon@samsung.com>
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ */
+#include <ini_interpreter.h>
+
+#include <vector>
+
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <parse_util.h>
+#include <util_func.h>
+
+#if defined(ENABLE_NNSTREAMER_BACKBONE)
+#include <nnstreamer_layer.h>
+#endif
+
+#if defined(ENABLE_TFLITE_BACKBONE)
+#include <tflite_layer.h>
+#endif
+
+static constexpr const char *FUNC_TAG = "[IniInterpreter] ";
+
+static constexpr const char *UNKNOWN_STR = "UNKNOWN";
+static constexpr const char *NONE_STR = "NONE";
+static constexpr const char *MODEL_STR = "model";
+static constexpr const char *DATASET_STR = "dataset";
+static constexpr const char *OPTIMIZER_STR = "optimizer";
+
+namespace nntrainer {
+
+namespace {
+
+/** @todo:
+ *  1. deprecate tag dispatching along with #1072
+ *  2. deprecate getMergeableGraph (extendGraph should accept graph itself)
+ */
+class PlainLayer {};    /**< Plain Layer tag */
```

Comment:
Ary you going to use plainlayer ( others layers ) and backbonelayer??

Reply:
Currently this is split because of #1072, it must be merged after #1072 is resolved.